src/src/lms/lms.service.ts

```diff
@@ -0,0 +1,435 @@
+import { LSMTestTrackingDto } from './dto/lms-test-tracking.dto';
+import { ConfigService } from '@nestjs/config';
+import { SearchLMSDto } from './dto/search-lms.dto';
+
+@Injectable()
+export class LMSService {
+	public table = 'events';
+	public fillable = [
+		'context',
+		'context_id',
+		'created_by',
+		'end_date',
+		'end_time',
+		'location',
+		'location_type',
+		'start_date',
+		'start_time',
+		'updated_by',
+		'user_id',
+	];
+	public returnFields = [
+		'id',
+		'name',
+		'context',
+		'context_id',
+		'created_by',
+		'end_date',
+		'end_time',
+		'type',
+		'location',
+		'master_trainer',
+		'location_type',
+		'start_date',
+		'start_time',
+		'updated_by',
+		'user_id',
+		'reminders',
+	];
+
+	public attendanceReturnFields = [
+		'id',
+		'user_id',
+		'context_id',
+		'created_by',
+		'context',
+		'status',
+		'lat',
+		'long',
+		'rsvp',
+		'photo_1',
+		'photo_2',
+		'date_time',
+		'updated_by',
+	];
+	constructor(
+		private readonly hasuraService: HasuraService,
+		private hasuraServiceFromServices: HasuraServiceFromServices,
+		private readonly userService: UserService,
+	) {}
+
+	public async getTestAllowStatus(header, response) {
+		const userDetail: any = await this.userService.ipUserInfo(header);
+		if (!userDetail?.data?.id) {
+			return response.status(400).send({
+				success: false,
+				message: 'Invalid User',
+				data: {},
+			});
+		} else {
+			let user_id = userDetail.data.id;
+
+			try {
+				let query = `query GetLms_test_tracking {
+				Lms_test_tracking(
+				  where:{
+					user_id:{
+					  _eq: "${user_id}"
+					}
+				  }
+				){
+					id
+				}
+			}`;
+				let data_list = await this.hasuraService.getData({ query });
+				if (data_list?.data?.Lms_test_tracking.length > 0) {
+					return response.status(200).send({
+						success: true,
+						message: 'Get Test Allow Status',
+						allowTest: false,
+					});
+				} else {
+					return response.status(200).send({
+						success: true,
+						message: 'Get Test Allow Status',
+						allowTest: true,
+					});
+				}
+			} catch (error) {
+				console.log(
+					`GetLms_test_tracking_allow_status '. Error!\n`,
+					error,
+					error.stack,
+				);
+				return response.status(404).send({
+					success: false,
+					message: 'Error in GetLms_test_tracking_allow_status!',
+					error: error,
+				});
+			}
+		}
+	}
+
+	public async createTestTracking(
+		lmsTestTrackingDto: LSMTestTrackingDto,
+		header,
+		response,
+	) {
+		const userDetail = await this.userService.ipUserInfo(header);
+		let user_id = userDetail.data.id;
+		lmsTestTrackingDto.user_id = user_id;
+		lmsTestTrackingDto.created_by = user_id;
+
+		let query_user_test = `query GetLms_test_tracking {
+			Lms_test_tracking(
+			  where:{
+				user_id:{
+				  _eq: "${user_id}"
+				}
+			  }
+			){
+				id
+			}
+		}`;
+		let data_list = await this.hasuraService.getData({
+			query: query_user_test,
+		});
+		if (data_list?.data?.Lms_test_tracking.length > 0) {
```

Comment:
@snehal0904 how we decide multiple attempts?
is this right place to decide that?

Reply:
where is test id?

Reply:
We have added the check here because we don't want to place 2 records for the same user.
@rushi-tekdi For this implementation, we have only one test ID but don't assume this, add test_id in the condition

Reply:
added test id in where condition